managed/services/server/server.go

```diff
@@ -550,17 +550,30 @@ func (s *Server) validateChangeSettingsRequest(ctx context.Context, req *serverp
 		return status.Error(codes.FailedPrecondition, "DBaaS is enabled via ENABLE_DBAAS or via deprecated PERCONA_TEST_DBAAS environment variable.")
 	}
 
-	if metricsRes.GetHr().AsDuration() != 0 && s.envSettings.MetricsResolutions.HR != 0 {
+	if metricsRes.GetHr().AsDuration() != 0 &&
+		s.envSettings.MetricsResolutions.HR != 0 &&
+		metricsRes.GetHr().AsDuration() != s.envSettings.MetricsResolutions.HR {
```

Comment:
It looks like it can be simplified to
```suggestion
	if s.envSettings.MetricsResolutions.HR != 0 && metricsRes.GetHr().AsDuration() != s.envSettings.MetricsResolutions.HR {
```

Reply:
Yes, that's correct. Logically it is the same condition.
I usually prefer to be more explicit in the conditions because it's easier to understand when you get back to the code later. You don't have to think twice - "and what about this case?" :)

Reply:
actually, it's not correct.
let's imagine the case that we send a request with HR = 0 and we have env setting = 1.
So in case of new changes, it will return an error, but without these changes, it won't.
Please revert your latest changes.

Reply:
at the beginning there was comparison like
`0 != 0 and 1 != 0 && 1 != 0`
now it's like `1 != 0 && 1 != 0`
which is completely different

Reply:
> actually, it's not correct. let's imagine the case that we send a request with HR = 0 and we have env setting = 1. So in case of new changes, it will return an error, but without these changes, it won't. Please revert your latest changes.

`HR = 0` means `unset`?

Reply:
yes

Reply:
Ok, got it